optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,220 @@
+import static io.camunda.optimize.service.util.configuration.EnvironmentPropertiesConstants.HTTPS_PORT_KEY;
+import static io.camunda.optimize.service.util.configuration.EnvironmentPropertiesConstants.HTTP_PORT_KEY;
+
+import io.camunda.optimize.rest.HealthRestService;
+import io.camunda.optimize.rest.LocalizationRestService;
+import io.camunda.optimize.rest.UIConfigurationRestService;
+import io.camunda.optimize.rest.constants.RestConstants;
+import io.camunda.optimize.rest.security.cloud.CCSaasAuth0WebSecurityConfig;
+import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
+import io.camunda.optimize.service.util.PanelNotificationConstants;
+import io.camunda.optimize.service.util.configuration.ConfigurationService;
+import io.camunda.optimize.service.util.configuration.EnvironmentPropertiesConstants;
+import io.camunda.optimize.tomcat.ExternalApiRewriteFilter;
+import io.camunda.optimize.tomcat.ExternalHomeServlet;
+import io.camunda.optimize.tomcat.OptimizeResourceConstants;
+import io.camunda.optimize.tomcat.ResponseHeadersFilter;
+import io.camunda.optimize.tomcat.URLRedirectFilter;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRegistration;
+import java.net.InetAddress;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import org.apache.catalina.connector.Connector;
+import org.apache.coyote.http2.Http2Protocol;
+import org.apache.tomcat.util.net.SSLHostConfig;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate.Type;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.embedded.tomcat.TomcatConnectorCustomizer;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+@Configuration
+public class TomcatConfig {
+  private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(Main.class);
+  private static final String COMPRESSED_MIME_TYPES =
+      "application/json," + "text/html," + "application/x-font-ttf," + "image/svg+xml";
+  private static final String PROTOCOL = "http/1.1";
+  private static final String LOGIN_ENDPOINT = "/login";
+  private static final String METRICS_ENDPOINT = "/metrics";
+  private static final String URL_BASE = "/#";
+
+  public static final String EXTERNAL_SUB_PATH = "/external";
+  public static final String ALLOWED_URL_EXTENSION =
+      String.join(
+          "|",
+          new String[] {
+            "/",
```

Comment:
I had to add this, otherwise Tomcat would not validate the home page, resulting into an infinite loop of redirections to the home page.

Reply:
I really don't know. It's been a puzzle